app/src/main/java/com/keylesspalace/tusky/ComposeActivity.java

```diff
@@ -1569,6 +1571,28 @@ private void saveDraftAndFinish() {
                         Log.e(TAG, String.format("Autocomplete search for %s failed.", token));
                         return Collections.emptyList();
                     }
+                case ':':
+                    try {
+                        emojiListRetrievalLatch.await();
```

Comment:
I'm not sure we need to wait for latch. Autocomplete should be fast and if it doesn't work then maybe it makes sense to just skip. Did you notice the change when you used it?

Reply:
The latch is there so that if you begin to type an emoji code and then stop, the result will always be consistent. So you'll never have to type an additional character and then delete it to see the results you want, if you're using autocomplete as a search.

The case where the `await` doesn't return immediately is out of the ordinary, and I have a fast connection, so I don't notice it in practice. I did test the feature with a throttled connection, and it works as expected.

I could remove the latch without affecting most potential users. I just figure it's a pretty small difference in code for some nice consistency.

Reply:
Yeah, maybe it's for the best there